### PR TITLE
add test mechanism and simple test for project #21

### PR DESCRIPTION
**Pull request by Oksana1788**

### CODE REVIEW

**Comment by Oksana1788** [comment]
Hi @yurii-polishchuk  and @anMelnyk  please check it. Thanks.

File: src/test/java/com/ipcount/core/ApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Autowired
+    private ClientController controller;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+
+    @Test
+    public void notNullTest() throws Exception {
```
Comment (anMelnyk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: @Oksana1788  please remove this test, as it doesn't have any practical value:))

File: src/test/java/com/ipcount/core/ApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ApplicationTest {
+
+    @Autowired
+    private ClientController controller;
```
Comment (anMelnyk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: @Oksana1788 please remove controller field. Also try to avoid field injection- use constructor injection instead.

File: src/test/java/com/ipcount/core/ApplicationTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void notNullTest() throws Exception {
+        assertThat(controller).isNotNull();
+    }
+
+    @Test
+    public void checkResponseBody() throws Exception {
+        this.mockMvc.perform(get("/clients")).andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString("[{\"id\":12,\"statusP\":5,\"tPlane\":8,\"speedIn\":5,\"speedOut\":5,\"ipAddress\":\"45\",\"name\":\"8\",\"phone\":\"9\",\"streetName\":\"5\",\"houseNumber\":\"5\",\"flatNumber\":\"5\",\"mail\":\"5\",\"password\":\"5\",\"sms\":\"q\",\"groupC\":5,\"speedLoc\":5,\"tools\":\"5\"}]")));
```
Comment (anMelnyk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: @Oksana1788 it looks like you use actual database for testing. Please use embedded one (h2) instead. Use Before annotation to add testing data to db.

File: src/test/resources/application-test.yml [hunk 4]
```diff
@@ -9,4 +9,8 @@ spring:
     driverClassName: org.h2.Driver
```
Comment (anMelnyk) [inline src/test/resources/application-test.yml]: @Oksana1788 it's great that you add test profile, but U never use it. Please check link https://docs.spring.io/spring/docs/current/spring-framework-reference/html/integration-testing.html . Take a look at ActiveProfile section.

File: pom.xml [hunk 5]
```diff
@@ -23,6 +23,12 @@
             <version>1.5.2.RELEASE</version>
         </dependency>
         <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (anMelnyk) [inline pom.xml]: @Oksana1788 please check if spring-boot-starter-test contains junit. If so, please remove junit dependency.

File: src/test/java/com/ipcount/core/ApplicationTest.java (hunk 3 above)
Comment (yurii-polishchuk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: that's not a problem, but try to make readable lines, maybe it a good idea to concatenate the long string or put it to separate readable value, etc.

File: src/test/java/com/ipcount/core/ApplicationTest.java [hunk 6]
```diff
@@ -28,68 +29,40 @@
...
-        clientEntity.setStreetName("1");
-        clientEntity.setHouseNumber("1");
-        clientEntity.setFlatNumber("1");
-        clientEntity.setMail("1");
-        clientEntity.setPassword("1");
-        clientEntity.setSms("1");
-        clientEntity.setGroupC(1);
-        clientEntity.setSpeedLoc(1);
-        clientEntity.setTools("1");
+        ClientBuilder.buildEntity(clientEntity);
```
Comment (yurii-polishchuk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: a builder should create new Entity and not consume existing one. Please search for Builder Pattern Example.

File: src/test/java/com/ipcount/core/ApplicationTest.java [hunk 7]
```diff
@@ -28,68 +29,40 @@
...
         clientRepository.save(clientEntity);
     }
 
 
     @Test
     public void checkResponseBody() throws Exception {
+        String a= "[{\"id\":1,\"statusP\":1,\"tPlane\":1,\"speedIn\":1,\"speedOut\":1,\"ipAddress\":\"1\",\"name\":\"1\",\"phone\":\"1\",\"streetName\":\"1\",\"houseNumber\":\"1\",\"flatNumber\":\"1\",\"mail\":\"1\",\"password\":\"1\",\"sms\":\"1\",\"groupC\":1,\"speedLoc\":1,\"tools\":\"1\"}]";
```
Comment (yurii-polishchuk) [inline src/test/java/com/ipcount/core/ApplicationTest.java]: I've meant just to split the string)) so it could be human readable.

**Comment by Oksana1788** [comment]
@anMelnyk  @yurii-polishchuk please check it.

File: src/test/java/com/ipcount/core/util/ClientBuilder.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
+package com.ipcount.core.util;
+
+import com.ipcount.core.entity.ClientEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+
+import java.util.Date;
+
+@Repository
```
Comment (yurii-polishchuk) [inline src/test/java/com/ipcount/core/util/ClientBuilder.java]: Why did you need this annotation on builder?

File: src/test/java/com/ipcount/core/util/ClientBuilder.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+public class ClientBuilder {
+
+    public ClientBuilder() {
+    }
+
+    public static ClientEntity builderE() {
+        ClientEntity clientEntity = new ClientEntity();
```
Comment (yurii-polishchuk) [inline src/test/java/com/ipcount/core/util/ClientBuilder.java]: you are going to create two object, this one and the second in builder(). Why do you need such approach?

Comment (Oksana1788) [inline src/test/java/com/ipcount/core/util/ClientBuilder.java]: Fixed
